File: src/botHandler/bots/arboretum/python/LLM_randomMessage/readLine.py
import json
import sys
from handleMove import handleMove


def eprint(*args, **kwargs):
    """Prints to stderr."""
    print(*args, file=sys.stderr, **kwargs)


def sendMove(move, messageID):
    """Sends the move to the game server."""
    output = {"move": move, "messageID": messageID}
    print(json.dumps(output))


def handleNewGame(gameNumber, messageID):
    response = {"result": "accepted", "gameNumber": gameNumber}
    sendMove(response, messageID)


def beginReadLine():
    while True:
        console_input = input()
        try:
            ## if the input is empty, continue to the next iteration
            if not console_input.strip():
                continue

            ## parse the input as JSON
            data = json.loads(console_input)
            state, messageID = data.get("state"), data.get("messageID")

            ## handle new game and end game state
            if state.message == "NEWGAME":
                handleNewGame(state)
                continue

            ## handle not active turn
            if not state.activeTurn:
                eprint(f"Not active turn: {state.activeTurn} != {state.playerID}")
                sendMove(0, messageID)
                continue

            ## handle active turn
            move = handleMove(state)
            sendMove(move, messageID)

        except:
            eprint(f"Error: {sys.exc_info()[0]}")

File: src/AcerAces-bots/arboretum/strategic-pacifist/readLine.ts
import { handleMove } from './handleMove.js';
import readline from 'node:readline';

// Handles reading lines
export function beginReadline(): void {
	const rl = readline.createInterface({
		input: process.stdin,
		output: process.stdout,
	});

	rl.on('line', (line: string) => {
		try {
			const { state, messageID } = JSON.parse(line) as {
				state: any;
				messageID: string;
			};
			
			if (state.message === 'NEWGAME' || state.message === 'ENDGAME') {
				sendNEWGAME(state.gameNumber, messageID);
				return;
			}

			if (!state.activeTurn) {
				sendMove(0, messageID);
				return;
			}

			const move = handleMove(state);
			sendMove(move, messageID);
		} catch (err) {
			console.error(err);
		}
	});

	function sendMove(move: any, messageID: string): void {
		const message = JSON.stringify({ move, messageID });
		process.stdout.write(message + '\n');
	}

	function sendNEWGAME(gameNumber: number, messageID: string): void {
		const response = { result: 'accepted', gameNumber };
		sendMove(response, messageID);
	}
} 